### PR TITLE
[ui]  Create and delete resource [no-api implementation]

### LINKED ISSUE #374: Support Creation & Deletion of device resources
### What is the purpose of the feature?
Being able to create and delete resources directly in the UI.

### What specific challenge does this feature solve?
Easy to use.

### What alternatives did I consider?
None

### PR DESCRIPTION
**Pull request by kuby**
- Extended the developer manual
- New component ActionButton
- Bugfix in resource types and interfaces listing
- Refactored the resource update modal to be able to work as a create resource modal
- Resources list contains an Action button

### CODE REVIEW

**Review by ondrejtomcik — APPROVED** [review]

File: docs/guide/developing/dashboard.md [hunk 1]
```diff
@@ -66,11 +83,17 @@ $ npm run generate-language-files
 
 Now your translations are updated and ready to be used.
 
+### Default language
+
+Default language is set to be `en`. This configuration can be overridden in `./http-gateway/web/src/config.js` by changing the `defaultLanguage` field. Change this only to a language which is supported by the application (is present in the `supportedLanguages` list).
```
Comment (oskarbauer) [inline docs/guide/developing/dashboard.md]: The default language is set to be `en`. This configuration can be overridden in `./http-gateway/web/src/config.js` by changing the `defaultLanguage` field. Change this only to a language that is supported by the application (is present in the `supportedLanguages` list).

File: docs/guide/developing/dashboard.md [hunk 2]
```diff
@@ -66,11 +83,17 @@ $ npm run generate-language-files
 
 Now your translations are updated and ready to be used.
 
+### Default language
+
+Default language is set to be `en`. This configuration can be overridden in `./http-gateway/web/src/config.js` by changing the `defaultLanguage` field. Change this only to a language which is supported by the application (is present in the `supportedLanguages` list).
+
+After the first visit of the application from a browser, it will remember the current language state, so in order to see the `defaultLanguage` field to work, you should clear the `localStorage` entry `language` from the browser.
```
Comment (oskarbauer) [inline docs/guide/developing/dashboard.md]: After your first visit to the application from a browser, the application will remember the current language state. In order to change the `defaultLanguage` field and see the change, you should clear the `localStorage` entry `language` from the browser.

File: docs/guide/developing/dashboard.md [hunk 3]
```diff
@@ -66,11 +83,17 @@ $ npm run generate-language-files
...
+Default language is set to be `en`. This configuration can be overridden in `./http-gateway/web/src/config.js` by changing the `defaultLanguage` field. Change this only to a language which is supported by the application (is present in the `supportedLanguages` list).
+
+After the first visit of the application from a browser, it will remember the current language state, so in order to see the `defaultLanguage` field to work, you should clear the `localStorage` entry `language` from the browser.
+
 ## Branding
 
 ### Colors change
 
-All colors are defined in one `scss` file located in `./http-gateway/web/src/common/styles/colors.scss`. Changing one of the colors will have an impact on all parts of the application.
+All colors are defined in one `scss` file `./http-gateway/web/src/common/styles/colors.scss`. Changing one of the colors will have an impact on all parts of the application.
```
Comment (oskarbauer) [inline docs/guide/developing/dashboard.md]: All colors are defined in one `scss` file: `./http-gateway/web/src/common/styles/colors.scss`. Changing one of the colors will have an impact on all parts of the application.

File: docs/guide/developing/dashboard.md [hunk 4]
```diff
@@ -89,3 +112,25 @@ Small logo (when the menu is collapsed):
 Favicon:
 
 - `./http-gateway/web/public/favicon.png`
+
+You might need to adjust some CSS in order to have the Logo rendered correctly, if the size is different than the recommended one. You can modify these values in `./http-gateway/web/src/components/left-panel/left-panel.scss`, look for the classes `.logo-big` and `.logo-small`. Adjust the height in these classes to fit your needs.
+
+### Application name
+
+Change the application name which also appears as a part of the Title bar in the browser header in file `./http-gateway/web/src/config.js` by changing the `appName` field.
```
Comment (oskarbauer) [inline docs/guide/developing/dashboard.md]: The application name which also appears in the title bar can be changed by modifying the `appName` field in `./http-gateway/web/src/config.js`

File: docs/guide/developing/dashboard.md [hunk 5]
```diff
@@ -89,3 +112,25 @@ Small logo (when the menu is collapsed):
...
+
+You might need to adjust some CSS in order to have the Logo rendered correctly, if the size is different than the recommended one. You can modify these values in `./http-gateway/web/src/components/left-panel/left-panel.scss`, look for the classes `.logo-big` and `.logo-small`. Adjust the height in these classes to fit your needs.
+
+### Application name
+
+Change the application name which also appears as a part of the Title bar in the browser header in file `./http-gateway/web/src/config.js` by changing the `appName` field.
+
+### Header / Status Bar
+
+Header has a dedicated component which can be found in `./http-gateway/web/src/components/status-bar/status-bar.js`. You can modify the `div` whith `id` of `status-bar` by removing the already present component like `LanguageSwitcher` or `UserWidget`, or simply adding other content next to these components.
```
Comment (oskarbauer) [inline docs/guide/developing/dashboard.md]: The header has a dedicated component which can be found in `./http-gateway/web/src/components/status-bar/status-bar.js`.

Comment (oskarbauer) [inline docs/guide/developing/dashboard.md]: You can modify the `div` with `id` of `status-bar` by removing the already present component like `LanguageSwitcher` or `UserWidget`, or simply adding other content next to these components.

Don't understand this.

File: docs/guide/developing/dashboard.md [hunk 6]
```diff
@@ -89,3 +112,25 @@ Small logo (when the menu is collapsed):
...
+
+### Footer
+
+Footer has a dedicated component which can be found in `./http-gateway/web/src/components/footer/footer.js`. You can modify the `footer` tag by removing the already present links, or simply adding other content next to them.
```
Comment (oskarbauer) [inline docs/guide/developing/dashboard.md]: Footer has a dedicated component which can be found in `./http-gateway/web/src/components/footer/footer.js`. You can modify the `footer` tag by removing the already present links, or simply adding different content next to them.

File: docs/guide/developing/dashboard.md [hunk 7]
```diff
@@ -89,3 +112,25 @@ Small logo (when the menu is collapsed):
...
+
+### Text changes
+
+Every text in this application is comming from a translation file located in `./http-gateway/web/src/languages/langauges.json`. This object contains as many blocks, as many languages you support in your application. If some of the blocks are missing you can simply duplicate them and use the same language code you used when you added a new language.
```
Comment (oskarbauer) [inline docs/guide/developing/dashboard.md]: Every text in this application is coming from a translation file located in `./http-gateway/web/src/languages/langauges.json`. This object contains a language block for each language you support in your application. If a block is missing you can duplicate an existing block and modify the block with the language code that is missing.

File: docs/guide/developing/dashboard.md [hunk 8]
```diff
@@ -89,3 +112,25 @@ Small logo (when the menu is collapsed):
...
+
+Some messages might be missing. This is due to fact that they were not yet translated. You can do that manually or automatically with some language manager like [POEditor](https://poeditor.com/).
```
Comment (oskarbauer) [inline docs/guide/developing/dashboard.md]: Some messages might be missing. This is due to fact that they were not yet translated. You can add them manually or use a language editor like [POEditor](https://poeditor.com/).

File: docs/guide/developing/dashboard.md [hunk 9]
```diff
@@ -89,3 +112,25 @@ Small logo (when the menu is collapsed):
...
+
+You can also override these strings to fit your need, for example if you would like to have `Devices` as a name of the menu which we call `Things`, you have to find the key `menu.things` and change its value to `Devices`.
```
Comment (oskarbauer) [inline docs/guide/developing/dashboard.md]: You can also override these strings to fit your need, for example, if you would like to have `Devices` as the name of the menu which we call `Things`, you have to find the key `menu.things` and change its value to `Devices`.

File: http-gateway/web/src/components/language-switcher/language-switcher-i18n.js [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import { defineMessages } from '@formatjs/intl'
+
+export const messages = defineMessages({
+  en: {
+    id: 'language-switcher.english',
```
Comment (oskarbauer) [inline http-gateway/web/src/components/language-switcher/language-switcher-i18n.js:5]: not using the id: 'language-switcher.en',?

**Review by oskarbauer — CHANGES_REQUESTED** [review]
There are some misspellings and some things that should be changed in the sentence structure of the .md doc.  With those changes, I approve.

File: docs/guide/developing/dashboard.md (hunk 5 above)
Comment (oskarbauer) [reply docs/guide/developing/dashboard.md]: You can modify the status-bar `<div id="status-bar">...</div>` by removing existing components like LanguageSwitcher and UserWidget or by adding different content in the div.

File: http-gateway/web/src/components/language-switcher/language-switcher-i18n.js (hunk 10 above)
Comment (kuby) [reply http-gateway/web/src/components/language-switcher/language-switcher-i18n.js:5]: Want to keep the `key` as the language code and the `id` used in the translation as the whole meaning of the translation string.

**Review by oskarbauer — APPROVED** [review]

**Pull request merged**